docs/source/model_design/vivarium_model_components/risk_factors/risk_mediation/index.rst

```diff
@@ -30,4 +30,399 @@ Risk factor mediation
+with the adjusted RR method ("Comparing crude RR versus mediator-adjusted RR"
+in `the GBD risk factors methods appendix [supplementary appendix 1] <https://www.thelancet.com/journals/lancet/article/PIIS0140-6736(20)30752-2/fulltext#supplementaryMaterial>`_) or
+the delta method ("Estimating the mediation factor by pathway of the effect").
+
+.. todo::
+
+  Where should someone look to find this out?
+
+Adjusted RR method
+^^^^^^^^^^^^^^^^^^
+
+In this case, GBD's data were a pair of RRs for the risk-outcome pair,
+one adjusted for the mediator and one not.
+We presume that the unadjusted RR is (approximately) equal to the total RR
+GBD reports for the risk-outcome pair.
+Therefore, we manipulate this equation from the GBD methods appendix:
+
+.. math::
+
+  MF = \frac{RR_\text{r→o,unadjusted for m} - RR_\text{r→o,adjusted for m}}{RR_\text{r→o,unadjusted for m} - 1}
+
+to solve for :math:`RR_\text{r→o,adjusted for m}`:
+
+.. math::
+
+  RR_\text{r→o,adjusted for m} = RR_\text{r→o,total} - MF * (RR_\text{r→o,total} - 1)
+
+:math:`RR_\text{r→o,adjusted for m}` is the underlying data point, but it
+isn't particularly useful to us, as it doesn't necessarily show up
+anywhere in our causal diagram.
+We can use it to calculate a delta, an (assumed linear) increase in the mediator
+per unit increase in the mediated risk:
+
+.. math::
+
+  \Delta_\text{r→m} = \frac{log(RR_\text{r→o,total}) - log(RR_\text{r→o,adjusted for m})}{log(RR_\text{m→o,total})}
+
+Every mediation pathway that begins with the same primary risk
+and mediator will give us the information we need to calculate such a linear
+risk-risk effect.
+For example, BMI→SBP→IHD and BMI→SBP→Stroke both imply delta values for BMI→SBP.
+However, the calculated values from different mediation pathways
+may be inconsistent, since GBD does not enforce consistency.
+We can only use one value in our simulation;
+for future models, we recommend averaging all the values,
+or averaging a subset of values that we believe are informed by higher-quality data.
+In the past, we have sometimes chosen an arbitrary pathway as the source
+of the simulation value.
+
+Once we select a single delta for each risk-mediator pair,
+these deltas are used to inform the risk to mediator arrows in our causal diagram.
+The only thing that remains is to find the RR of the direct effect:
+
+.. math::
+
+  RR_\text{r→o,direct} = \frac{RR_\text{r→o,total}}{\prod_{m \in M} (RR_\text{m→o})^{\Delta_\text{r→m}}}
+
+.. note::
+  We may calculate a direct RR below 1.
+  This is not logically incoherent (a risk factor could have a protective direct effect
+  that is counteracted by harmful indirect/mediated effects), but it may
+  also indicate problems in the GBD mediation estimation, particularly if a
+  protective effect doesn't make sense given domain knowledge about the
+  risk factor.
+
+  We'll deal with these situations on a case-by-case basis.
+
+Delta method
+^^^^^^^^^^^^
```

Comment:
Just a note if I am remembering correctly -- I think that @SylLutze tested what we would get for delta for various mediation pathways in CVD under each of these methods and found that they were really close to one another. It might be worth noting that somewhere (if that was actually true) given that we so far don't really know how/where to learn about which method GBD used for specific pathway.

Reply:
@SylLutze I'd be interested to hear more about this! I have not actually tried doing this for real mediation factors reported by GBD so don't have a good sense.

Reply:
Hey - yeah did calculate the deltas a few times. 
1. In the back calculate then forward calculate math. The deltas were not as internally consistent as you would want actually! For example BMI -> SBP -> IHD and BMI -> SBP -> Stroke. In theory, BMI -> SBP should be the same delta in both pathways, but we did not find this to be true. 
2. We also checked deltas in regards to an intervention. We had an intervention that affected FPG (treat -> FPG) and affected BMI (treat -> BMI -> FPG) and used deltas to see if we needed to include the (treat -> FPG) pathway, or if it was mainly due to BMI mediation. We found that it had a significant direct effect. 

I was having trouble finding where this comment chain started in the text so let me know if this is unclear/irrelevant.

Reply:
Right -- that is all sounding familiar now. Thanks for the reminder.

I'm curious -- did you try back-calculation using both the delta method and the adjusted RR method? The math is slightly different:

$$\Delta_\text{r→m} = \frac{\log(RR_\text{r→o,total}) - \log(RR_\text{r→o,total} - MF \times (RR_\text{r→o,total} - 1))}{\log(RR_\text{m→o,total})}$$

when GBD used adjusted RR;

$$\Delta_\text{r→m} = \frac{MF \times (RR_\text{r→o,total} - 1) + 1}{\log(RR_\text{m→o,total})}$$

when GBD used the delta method.

I think Ali's question is whether we end up getting similar results with either equation for actual values of RRs and MFs.

Reply:
Oh - thanks for clarifying. I used the bottom equation for testing (although it looks like I took the log of the top part as well). It wouldn't be too difficult in the existing notebook to add the top equation and see how similar or different the results are. Would you like me to add that and see?

Reply:
> although it looks like I took the log of the top part as well

Hmm, let me double check my algebra.

Can you point me to your implementation of this?

Reply:
Here's the file where I do mediation things. https://github.com/ihmeuw/vivarium_research_nih_us_cvd/blob/main/data_testing_exploration/risk_mediation_2.ipynb

It has some comments but maybe not the best, so happy to go over it together in a coworking or something if you want.

Reply:
Thanks! Yes, the top of the fraction should be inside a log. I'll correct this in the other PR. I've also corrected it in the comment above if that is easier to use as a reference.

I think a comparison of the two equations would be great! Your notebook won't immediately run for me so I will probably need your help with this.

Reply:
Hey hey - I added the new equation to the notebook. It seems like the top equation gives a delta that is consistently lower than the bottom equation. The average difference is around ~0.1. I tested for measures: BMI -> FPG -> IHD, BMI -> FPG -> stroke. And the same two pathways with SBP instead of FPG. Updated notebook: https://github.com/ihmeuw/vivarium_research_nih_us_cvd/blob/main/data_testing_exploration/risk_mediation_2.ipynb

Also the notebook should run now for the sections you need. Sorry about that - some lines were commented out to have the by draw sections run and it wasn't well documented. 

Lemme know if you want to connect about any of this!

Reply:
Interesting, thanks for doing this!

I think my takeaway from your results is: it does matter a decent amount which equation you use, so it is worth trying to find out which way GBD did the calculation. I currently have a TODO for looking into how to find out which they did, so I think I'll leave that TODO as is. Sound right?

Reply:
Hmm interesting! We can note this but it doesn't seem worth a model rerun to switch equations. 

The TODO sounds right to me. Sounds like you might have figured out how to see what GBD did though? Curious to hear more at the next docs session.

Reply:
Yeah, definitely not worth a rerun.

Nope -- they give a few examples, and those happen to be some of them. But I haven't yet found a full list.